_posts/2018-11-20-brief_introduction_for_spring_batch.markdown

```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title: A brief introduction of Spring-Batch in theory
+date: 2018-11-20 14:20:22 +0800
+tags: [JAVA, Archiver, Spring-Batch]
+---
+
+This article will briefly introduce the background, scenarios of Spring-Batch and principles of designing a general batch.  
+Also this article can be treat as a reading review of the Spring Batch - Reference Documentation, All I want to find out is “Whether Archiver is suitable for this so called industry standard framework?” 
+
+#### **BACKGROUND**
+Many applications within the enterprise require bulk processing to perform business operations. However there has been a lack of such a Java-based batch processing framework for years. So springSource and Accenture collaborated, try to standardized this area. 
+In RingCentral the Archiver is also a bulk processing system, to figure out whether Archiver is suitable we should know what kind of features that Spring-Batch supply. 
+
+#### **SCENARIOS**
+###### Business Scenarios 
+ 	•	Periodically : allow to be execute periodically 
+ 	•	Concurrent : can be process in parallel lines 
+ 	•	Staged, enterprise message-driven processing
+ 	•	Massively : may have big amount of data to process 
+ 	•	Manual or scheduled : have multiple way to trigger 
+ 	•	Step by Step : sequential processing of dependent steps 
+ 	•	Partial processing: allow skip records 
+ 	•	Whole-batch transaction, for cases with a small batch size or existing stored procedures/scripts
+  
+###### Technical Objectives
+ 	•	Allow developer concentrate on business logic and let the framework take care of infrastructure 
+ 	•	Separate the infrastructure, the batch execution environment 
+ 	•	Provide common, core execution services as interfaces that all projects can implement 
+ 	•	Provide simple and default implementations of the core execution interfaces that can be used ‘ootb’(out of the box) 
+ 	•	Easy to configure, customize, and extend services, by leveraging the spring framework in all layers. 
+ 	•	All existing core services should be easy to replace or extend, without any impact to the infrastructure layer. 
+ 	•	Provide a simple deployment model, with the architecture JARs completely separate from the application 
+As above, most of the features matches Archiver. So my conclusion is we can get familiar with this framework first, and use it in some new jobs. In the nearly future if this is much more easier to maintain and it has some optimized database / IOPS practice as it says in the document. We can use it entirely and directly. 
```

Comment:
As shown above, or as explained above, just 'as above' makes no sense.
Also, 'most of the features match Archiver's needs'.
So my conclusion is 'that' we can ... 
In the 'near' future... near is adjective and future is noun. Nearly is adverb that you use to describe some action usually, like 'nearly done' or 'I nearly bumped into a tree'. 
In the near future, if this is much easier (much more easier is bad) to maintain and has some optimized database / IOPS practices as claimed in the document, we can use it directly (entirely is kind of weird and redundant here).